compiler/next/test/frontend/testParseUse.cpp

```diff
@@ -0,0 +1,244 @@
+  assert(use->visibility() == Decl::PRIVATE);
+  assert(use->numUseClauses() == 1);
+  const UseClause* useClause = use->useClause(0);
+  assert(useClause->name()->isAs());
+  const As* as = useClause->name()->toAs();
+  assert(as->name()->isIdentifier());
+  assert(as->name()->toIdentifier()->name() == "A");
+  assert(as->rename()->isIdentifier());
+  assert(as->rename()->toIdentifier()->name() == "X");
+  assert(useClause->limitationClauseKind() == UseClause::EXCEPT);
+  assert(useClause->numLimitations() == 3);
+  assert(useClause->limitation(0)->isIdentifier()); 
+  assert(useClause->limitation(1)->isIdentifier()); 
+  assert(useClause->limitation(2)->isIdentifier()); 
+
+  // Make sure the limitations iterator works as expected.
+  {
+    int count = 0;
+    for (const auto limitation : useClause->limitations()) {
+      assert(limitation->tag() == asttags::Identifier);
+      assert(limitation->tag() == useClause->limitation(count)->tag());
+      assert(limitation->isIdentifier());
+      count++;
+    }
+    assert(count == useClause->numLimitations());
+  }
+}
+
+static void test3(Parser* parser) {
+  auto parseResult = parser->parseString("test3.chpl",
+      "/*c1*/\n"
+      "use A.SM1 only Foo as X, Bar, Baz as Y;\n"
+      "/*c7*/\n");
+  assert(parseResult.errors.size() == 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModule());
+  auto mod = parseResult.topLevelExpressions[0]->toModule();
+  assert(mod->numStmts() == 3);
+  assert(mod->stmt(0)->isComment());
+  assert(mod->stmt(1)->isUse());
+  assert(mod->stmt(2)->isComment());
+  const Use* use = mod->stmt(1)->toUse();
+  assert(use->numUseClauses() == 1);
+  const UseClause* useClause = use->useClause(0);
+  assert(useClause->name()->isDot());
+  const Dot* dot = useClause->name()->toDot();
+  assert(dot->calledExpression()->isIdentifier());
+  assert(dot->calledExpression()->toIdentifier()->name() == "A");
+  assert(dot->field() == "SM1");
+  assert(useClause->limitationClauseKind() == UseClause::ONLY);
+  assert(useClause->numLimitations() == 3);
+  // Check the first limitation.
+  {
+    assert(useClause->limitation(0)->isAs());
+    const As* as = useClause->limitation(0)->toAs();
+    assert(as->name()->isIdentifier());
+    assert(as->name()->toIdentifier()->name() == "Foo");
+    assert(as->rename()->isIdentifier());
+    assert(as->rename()->toIdentifier()->name() == "X");
+  }
+  // Check the second limitation.
+  {
+    assert(useClause->limitation(1)->isIdentifier());
+    assert(useClause->limitation(1)->toIdentifier()->name() == "Bar");
+  }
+  // Check the third limitation.
+  {
+    assert(useClause->limitation(2)->isAs());
+    const As* as = useClause->limitation(2)->toAs();
+    assert(as->name()->isIdentifier());
+    assert(as->name()->toIdentifier()->name() == "Baz");
+    assert(as->rename()->isIdentifier());
+    assert(as->rename()->toIdentifier()->name() == "Y");
+  }
+}
```

Comment:
I don't see tests of `except *;` or `only;`.  I'm a little worried the check for limitations in the `only;` case is wrong.

Reply:
I will add some tests for those.

Can you explain what you mean (your fears) by your second sentence?

Reply:
Oh I see. I can add a test for `only;`. I expect the limitations list will be empty and the `limitationClauseKind()` will be `UseClause::ONLY`.

Reply:
Tests added, and fixed that bug. Thanks!